### PR TITLE
KAFKA-2863; Add  a `close()` method to `Authorizer`

### PR DESCRIPTION
**Pull request by ijuma**

### CODE REVIEW

**Comment by ijuma** [comment]
./gradlew test passed locally.

File: core/src/main/scala/kafka/security/auth/Authorizer.scala [hunk 1]
```diff
@@ -18,7 +18,7 @@
 package kafka.security.auth
 
 import kafka.network.RequestChannel.Session
-import org.apache.kafka.common.Configurable
+import org.apache.kafka.common.{KafkaException, Configurable}
```
Comment (junrao) [inline core/src/main/scala/kafka/security/auth/Authorizer.scala]: Is KafkaException used?

File: core/src/main/scala/kafka/security/auth/SimpleAclAuthorizer.scala [hunk 2]
```diff
@@ -226,6 +226,11 @@ class SimpleAclAuthorizer extends Authorizer with Logging {
     aclCache.toMap
   }
 
+  def close() {
+    zkUtils.zkClient.unsubscribeStateChanges(ZkStateChangeListener)
+    aclChangeListener.close()
```
Comment (junrao) [inline core/src/main/scala/kafka/security/auth/SimpleAclAuthorizer.scala]: It seems that we need to close zkUtils. If we do that, there is no need to unsubscribe here and in aclChangeListener.

File: core/src/main/scala/kafka/security/auth/Authorizer.scala (hunk 1 above)
Comment (ijuma) [inline core/src/main/scala/kafka/security/auth/Authorizer.scala]: There was a throws clause, but I removed that, so not anymore. Removed it locally.

File: core/src/main/scala/kafka/security/auth/SimpleAclAuthorizer.scala (hunk 2 above)
Comment (ijuma) [inline core/src/main/scala/kafka/security/auth/SimpleAclAuthorizer.scala]: Hmm, `zkUtils` will be used by other classes too and we close it in a different scope. Maybe I should leave `close()` as a `no-op` for `SimpleAclAuthorizer`?

Comment (ijuma) [inline core/src/main/scala/kafka/security/auth/SimpleAclAuthorizer.scala]: Sorry, ignore my comment. :) Will do as you say.

**Comment by ijuma** [comment]
Updated `ZkUtils.close()` and updated tests to close the authorizer. `gradlew test` passed locally. @junrao, ready for another review.

**Comment by junrao** [comment]
Thanks for the patch. LGTM

**Pull request closed without merging**